framework/src/meshgenerators/AdvancedExtruderGenerator.C

```diff
@@ -80,23 +96,65 @@ FancyExtruderGenerator::validParams()
+                                                              "Upward interface boundary ids.");
+
+  params.addParam<std::vector<std::vector<subdomain_id_type>>>(
+      "downward_boundary_source_blocks",
+      "Block ids used to generate downward interface boundaries.");
+
+  params.addParam<std::vector<std::vector<boundary_id_type>>>("downward_boundary_ids",
+                                                              "Downward interface boundary ids.");
+  params.addParamNamesToGroup(
+      "top_boundary bottom_boundary upward_boundary_source_blocks upward_boundary_ids "
+      "downward_boundary_source_blocks downward_boundary_ids",
+      "Boundary Assignment");
+  params.addParamNamesToGroup(
+      "subdomain_swaps boundary_swaps elem_integer_names_to_swap elem_integers_swaps", "ID Swap");
+
   return params;
 }
 
-FancyExtruderGenerator::FancyExtruderGenerator(const InputParameters & parameters)
+AdvancedExtruderGenerator::AdvancedExtruderGenerator(const InputParameters & parameters)
   : MeshGenerator(parameters),
     _input(getMesh("input")),
     _heights(getParam<std::vector<Real>>("heights")),
+    _biases(isParamValid("biases") ? getParam<std::vector<Real>>("biases")
+                                   : std::vector<Real>(_heights.size(), 1.0)),
     _num_layers(getParam<std::vector<unsigned int>>("num_layers")),
     _subdomain_swaps(getParam<std::vector<std::vector<subdomain_id_type>>>("subdomain_swaps")),
+    _boundary_swaps(getParam<std::vector<std::vector<boundary_id_type>>>("boundary_swaps")),
     _elem_integer_names_to_swap(getParam<std::vector<std::string>>("elem_integer_names_to_swap")),
-    _elem_integers_swaps(getParam<std::vector<std::vector<unsigned int>>>("elem_integers_swaps")),
+    _elem_integers_swaps(getParam<std::vector<std::vector<dof_id_type>>>("elem_integers_swaps")),
     _direction(getParam<Point>("direction")),
     _has_top_boundary(isParamValid("top_boundary")),
     _top_boundary(isParamValid("top_boundary") ? getParam<boundary_id_type>("top_boundary") : 0),
     _has_bottom_boundary(isParamValid("bottom_boundary")),
     _bottom_boundary(isParamValid("bottom_boundary") ? getParam<boundary_id_type>("bottom_boundary")
-                                                     : 0)
+                                                     : 0),
+    _upward_boundary_source_blocks(
+        isParamValid("upward_boundary_source_blocks")
+            ? getParam<std::vector<std::vector<subdomain_id_type>>>("upward_boundary_source_blocks")
+            : std::vector<std::vector<subdomain_id_type>>(_heights.size(),
+                                                          std::vector<subdomain_id_type>())),
+    _upward_boundary_ids(
+        isParamValid("upward_boundary_ids")
+            ? getParam<std::vector<std::vector<boundary_id_type>>>("upward_boundary_ids")
+            : std::vector<std::vector<boundary_id_type>>(_heights.size(),
```

Comment:
I dont like the default addition of a top and bottom boundary
was it like this before?

Reply:
looks like it :(

Reply:
Yes